docs/concept-workflows.md

```diff
@@ -1,30 +1,210 @@
+- Enable invocation from multiple other Workflow.
+- Workaround event history size limits.
+- Create one-to-one mappings between a Workflow Id and some other resource.
+- Execute some periodic logic.
+
+One of the main reasons you would not want to use a Child Workflow is the lack of a shared state with the Parent Workflow.
+A Parent and Child Workflow can communicate only through asynchronous signals.
+If the executing logic has tight coupling between Workflows, it may simply be easier to use a single Workflow that can rely on a shared object's state.
+
+## Example subscription Workflow use-case
+
+Let's look at a subscription based use-case to compare the difference between a Temporal application and other modern approaches.
+
+The basic business steps are as follows:
+
+1. A customer signs up for a service that has a trial period.
+2. After the trial period, if the customer has not cancelled, they should be charged once per month for up to x months.
+3. The customer has to be notified via email about the charges and should be able to cancel the subscription at any time.
+
+This business logic is not very complicated and can be expressed in a few dozen lines of code.
+But any practical implementation has to ensure that the business process is fault tolerant and scalable.
+
+### Design approach 1
+
+The first approach might be to center everything around a database where an application process would periodically scan the database tables for customers in specific states, execute necessary actions, and update the state to reflect that.
+
+But there are various drawbacks.
+
+1. The most obvious is that the application state machine of the customer's state quickly becomes extremely complicated.
+For example, if a credit card charge attempt fails or sending an email fails due to some a downstream system unavailability, the state is in a sort of limbo.
+2. Failed calls likely need to be retried for a long time, and these calls need to be throttled to not overload external resources.
+3. There needs to be logic to handle corrupted customer records to avoid blocking the whole process.
+4. Additionally databases have performance limitations and are not efficient for scenarios that require constant polling.
+
+### Design approach 2
+
+The next commonly employed approach is to use a timer service and queues.
+Updates are pushed to a queue and then some other service consumes then one at a time, updating a database, and possibly pushing more messages into other downstream queues.
+A timer service can be used to schedule queue polling or database actions.
+
+While this approach has shown to scale a bit better, the programming model can become very complex very quickly and is very prone to errors, as there are usually no transactional updates between a queuing system and a database.
+
+### Temporal design approach
+
+The Temporal programming model aims to encapsulate and implement the entire business logic in a simple function.
```

Comment:
I don't like "a simple function" emphasis. As it creates the impression that multiple functions or more complex OO-based workflows are not a thing.